File: javascript/challenges/280-love-is-all-we-have-the-only-way-that-each-can/index.test.ts
import { expect, describe, it } from "vitest";
import { replaceZeros } from ".";

// Talk is cheap, show me the code
describe.concurrent("Check errors", () => {
  it("test when input is empty", async () => {
    expect(() => replaceZeros("")).toThrowError(
      "Input must contain only digits."
    );
  });

  it("test when input doesn't contain only numbers", async () => {
    expect(() => replaceZeros("angel00123456")).toThrowError(
      "Input must contain only digits."
    );
  });
});

describe.concurrent("Check replaces", () => {
  it("test example 01", async () => {
    expect(replaceZeros("1234500362000440")).toBe("1234523623441");
  });

  it("test example 02", async () => {
    expect(replaceZeros("123450036200044")).toBe("123452362344");
  });

  it("test example 03", async () => {
    expect(replaceZeros("000000000000")).toBe("12");
  });

  it("test example 04", async () => {
    expect(replaceZeros("123456789")).toBe("123456789");
  });
});

File: ruby/challenges/280-love-is-all-we-have-the-only-way-that-each-can/index.test.rb
# frozen_string_literal: true

require 'minitest/autorun'
require_relative 'index'

# Talk is cheap, show me the code
describe 'ReplaceZerosErrorsTestSuite' do
  describe 'when try to replace empty string' do
    it 'must raise StandardError' do
      assert_raises(StandardError) { replace_zeros('') }
    end
  end

  describe "when input doesn't contain only numbers" do
    it 'must raise StandardError' do
      assert_raises(StandardError) { replace_zeros('angel00123456') }
    end
  end
end

describe 'ReplaceZerosTestSuite' do
  describe 'test example "1234500362000440"' do
    it 'must be 1234523623441' do
      assert replace_zeros('1234500362000440'), '1234523623441'
    end
  end

  describe 'test example "123450036200044"' do
    it 'must be 123452362344' do
      assert replace_zeros('123450036200044'), '123452362344'
    end
  end

  describe 'test example "000000000000"' do
    it 'must be 12' do
      assert replace_zeros('000000000000'), '12'
    end
  end

  describe 'test example "123456789"' do
    it 'must be 123456789' do
      assert replace_zeros('123456789'), '123456789'
    end
  end
end
